src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -61,17 +61,24 @@ public void Dispose()
                 return;
             }
 
-            int length = _utf8Json.Length;
-            _utf8Json = ReadOnlyMemory<byte>.Empty;
             _parsedData.Dispose();
 
-            // When "extra rented bytes exist" they contain the document,
-            // and thus need to be cleared before being returned.
-            if (_extraRentedBytes != null)
+            int length = _utf8Json.Length;
+            if (length > 0)
```

Comment:
> Added condition to validate _utf8Json.Length is > 0 in order to prevent that `extraRentedBytes.AsSpan(0, length).Clear()` takes a zero length range in a race condition.

@bartonjs, this looks like a good change to me. Was there a rationale for not checking (maybe perf - though its a single branch) or was it an oversight? I could imagine this being a rare enough for us to not guard against race condition resulting in the array not getting cleared.

Reply:
Oversight / assuming single-threaded execution.

The length variable could be moved up to line 59, and then the "IsEmpty" call replaced with `length == 0`; then the value isn't being double-dipped still.